File: DotNetSolution/Tests/DotNetTests/CliTests.cs
using CliProject;

namespace DotNetTests
{
    public class CliTests
    {
        private GreeterCli greeter;

        [SetUp]
        public void Setup()
        {
            greeter = new GreeterCli();
        }

        [Test]
        public void SayHello_ReturnsExpectedString()
        {
            var result = greeter.SayHello();
            Assert.That(result, Is.EqualTo("Hello from C++/CLI Project!"));
        }
    }
}

File: DotNetSolution/Tests/CppTests/CliTests.cpp
#include "pch.h"
#include "..\..\CliProject\GreeterCli.h"

namespace CppTests {
    TEST(CliTests, SayHello_ReturnsExpectedString) {
        GreeterCli greeter;

        std::string result = greeter.SayHello();

        EXPECT_EQ(result, "Hello from C++/CLI Project!");
    }
}